src/librustc_mir/interpret/operator.rs

```diff
@@ -135,12 +134,12 @@ impl<'a, 'mir, 'tcx, M: Machine<'mir, 'tcx>> EvalContext<'a, 'mir, 'tcx, M> {
                 ty,
             };
             match op {
-                Eq => PrimVal::from_bool(l.try_cmp(r).unwrap() == Ordering::Equal),
-                Ne => PrimVal::from_bool(l.try_cmp(r).unwrap() != Ordering::Equal),
-                Lt => PrimVal::from_bool(l.try_cmp(r).unwrap() == Ordering::Less),
-                Le => PrimVal::from_bool(l.try_cmp(r).unwrap() != Ordering::Greater),
-                Gt => PrimVal::from_bool(l.try_cmp(r).unwrap() == Ordering::Greater),
-                Ge => PrimVal::from_bool(l.try_cmp(r).unwrap() != Ordering::Less),
```

Comment:
Oof. `rustc_const_math::ConstFloat` in 1.26 doesn't `impl PartialOrd`. 
We either need to modify the [`try_cmp` method](https://github.com/Mark-Simulacrum/rust/blob/236755b27ab02c2d9ab0a8dcac3221c3934120b4/src/librustc_const_math/float.rs#L34-L58) or creates a new method which returns `Result<Option<Ordering>, ConstMathErr>`.

Reply:
How do you feel about https://github.com/rust-lang/rust/pull/51045/commits/7b58d76c98b06d7607c7eba848633d3f562bcbde? That looks fairly minimal and implements the required change to make the original patch work on stable AFAICt.